src/libsyntax/attr.rs

```diff
@@ -877,10 +886,38 @@ pub fn find_repr_attrs(diagnostic: &Handler, attr: &Attribute) -> Vec<ReprAttr>
                     if let Some(h) = hint {
                         acc.push(h);
                     }
+                } else if let Some((name, value)) = item.name_value_str() {
```

Comment:
Seems like you want to support only `#[repr(align = "16")]` but not `#[repr(align(16))]`?
@alexcrichton used to think otherwise, but I hope he can be pursued to only allow the latter form, *as it's already implemented* and thus likelier to be stabilized first. In fact, we can stabilize it for specific attributes (like this one) and leave the general case unstable.